src/main/scala/com/fulcrumgenomics/umi/UmiConsensusCaller.scala

```diff
@@ -252,7 +252,13 @@ trait UmiConsensusCaller[ConsensusRead <: SimpleRead] {
       var index = if (!rec.isFrPair) trimToLength - 1 else {
         // Get the number of mapped bases to clip that maps beyond the mate's end, including any soft-clipped bases. Use
         // that to compute where in the read to keep.
-        val clipPosition = rec.length - this.clipper.numBasesExtendingPastMate(rec=rec)
+        val mateUnclippedStart = rec.mateUnclippedStart.getOrElse(throw new IllegalStateException(f"Mate cigar (MC SAM tag) needed for read: ${rec.name}"))
+        val mateUnclippedEnd   = rec.mateUnclippedEnd.getOrElse(throw new IllegalStateException(f"Mate cigar (MC SAM tag) needed for read: ${rec.name}"))
```

Comment:
Given the `UmiConsensusCaller` doesn't really have access to the mate here, I think it makes sense to raise an exception as you have done. Mind adding this requirement to the scaladoc with a `@raises IllegalArgumentException`?

The exceptions should be for an illegal argument (correct type, bad value):

```suggestion
        val mateUnclippedStart = rec.mateUnclippedStart.getOrElse(throw new IllegalArgumentException(f"Mate cigar (MC SAM tag) needed for read: ${rec.name}"))
        val mateUnclippedEnd   = rec.mateUnclippedEnd.getOrElse(throw new IllegalArgumentException(f"Mate cigar (MC SAM tag) needed for read: ${rec.name}"))
```